components/ws-proxy/pkg/utils/rate-limited-listener.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package utils
```

Comment:
nit: `utils` is an actively discouraged package name (see [here](https://blog.golang.org/package-names), **Avoid meaningless package names**) ... maybe `ratelimit`

Reply:
Good point. Shame on me for being too lazy to find a proper package name. :laughing: